codalab/rest/users.py

```diff
@@ -1,10 +1,133 @@
 """
-Worksheets REST API Users Views
+Worksheets REST API Users Views.
+
+TODO(sckoo): Factor out the reusable parts of the JSON API views.
 """
-from bottle import get
+from bottle import get, route, response, request, local, HTTPResponse
+from marshmallow import ValidationError
+from marshmallow_jsonapi import Schema, fields
+
+from codalab.common import UsageError
+from codalab.lib import formatting
+from codalab.server.authenticated_plugin import AuthenticatedPlugin
+from codalab.server.json_api_plugin import JsonApiPlugin
+
+
+PUBLIC_USER_FIELDS = ('id', 'user_name', 'email', 'first_name', 'last_name',
+                      'date_joined', 'affiliation', 'url')
+
+
+class DataSize(fields.Field):
+    def _serialize(self, value, attr, obj):
+        return formatting.size_str(value)
+
+    def _deserialize(self, value, attr, data):
+        return formatting.parse_size(value)
+
+
+class Duration(fields.Field):
+    def _serialize(self, value, attr, obj):
+        return formatting.duration_str(value)
+
+    def _deserialize(self, value, attr, data):
+        return formatting.parse_duration(value)
+
+
+class UserSchema(Schema):
+    id = fields.String(dump_only=True, attribute='user_id')
+    user_name = fields.String()
+    email = fields.Email(dump_only=True)
```

Comment:
I disabled it for now until I've implemented the re-verification process (changing email should require verification of the new email address.)